### DISCUSSION TITLE
v2.2.0

### DISCUSSION THREAD

**Discussion by iSoron** [iSoron/uhabits#1927] [Announcements]
## Added
- Add support for Android 14 (@iSoron, @hiqua)
- Allow user to change app language (@leondzn)

## Fixed
- Implement workaround to make notifications non-dismissible in Android 14 (@iSoron, #1872)
- Fix splash screen background color in dark mode (@SIKV, #1888)

<hr /><em>This discussion was created from the release <a href='https://github.com/iSoron/uhabits/releases/tag/v2.2.0'>v2.2.0</a>.</em>

**Comment by iSoron** [comment]
**Update:** This release is still not available in Google Play because Google has rejected our update, claiming that the ability to set exact alarms (precisely-timed notifications) is not directly related to Loop Habit Tracker's core purpose. We have submitted an appeal.

**Reply by hiqua** [reply]
According to https://developer.android.com/about/versions/14/changes/schedule-exact-alarms#use-cases, the alternative would be to allow the reminders to appear within a 10-minute window using `setWindow`: https://developer.android.com/reference/android/app/AlarmManager#setWindow(int,%20long,%20long,%20android.app.PendingIntent).

**Reply by iSoron** [reply]
@hiqua If they deny the appeal, we might need to switch to this. We can also request the permission to set exact alarms during runtime, but Google is making it deliberately hard for apps to get it. Instead of showing the usual permission dialog with the "Allow / Deny" options, we can at most redirect the user to the alarms settings page. The user needs to manually click "Loop", then "Allow exact alarms", which is quite non-intuitive.

**Reply by iSoron** [reply]
@hiqua They accepted the appeal; we can continue using `USE_EXACT_ALARM`. I hope this doesn't get flagged again and again in the future.
